lib/services/cloud-projects-service.ts

```diff
@@ -0,0 +1,53 @@
+import { ExportOptions } from "../constants";
+
+export class CloudProjectsService implements ICloudProjectsService {
+	constructor(private $remoteProjectService: IRemoteProjectService,
+		private $errors: IErrors,
+		private $prompter: IPrompter) { }
+
+	public async getSolutionProjectInfo(opts: ISolutionProjectPairOptions): Promise<ISolutionProjectInfo> {
+		const apps = await this.$remoteProjectService.getAvailableAppsAndSolutions();
+		let framework = "";
+		let id = "";
+		const noSolutionNamePassed = !opts.solutionName;
+
+		if (noSolutionNamePassed) {
+			const solutionNames = apps.map(sln => sln.colorizedDisplayName) || [];
+			opts.solutionName = await this.$prompter.promptForChoice("Select solution to export", solutionNames);
+		}
+
+		const solutionData = await this.$remoteProjectService.getSolutionData(opts.solutionName);
+		if (!solutionData.Items || !solutionData.Items.length) {
+			this.$errors.failWithoutHelp(`Solution ${solutionData.Name} does not contain any projects.`);
+		}
+
+		if ((noSolutionNamePassed && !opts.projectName) || (opts.forceChooseProject && !opts.projectName)) {
+			const projects = (await this.$remoteProjectService.getProjectsForSolution(opts.solutionName)).map(proj => proj.Name);
+			if (opts.enableExportWholeSolution) {
+				projects.push(ExportOptions.WholeSolution);
+			}
+
+			const selection = await this.$prompter.promptForChoice("Select project to export", projects);
```

Comment:
same question for non-interactive terminal

Reply:
this could be a problem - will fix

Reply:
fixed with the latest commit